File: Csource/fan_motor_test3.c
#include <wiringPi.h>
#include <softPwm.h>
#include <stdio.h>
#include "motor.h" //내폴더에 있을땐 "" 라이브러리에 넣어두면 <>

int main (void) {

    if (wiringPiSetup() == -1) { return 1; }

    softPwmCreate(FAN_IA,0,1000);
    
    // 무한루프
    while (1) {
        Weak(); //미풍 5초
        Medium(); //약풍 5초
        Strong(); //강풍 5초
    }

    return 0;
}

File: Python/fan_motor_test3.py
import RPi.GPIO as GPIO
import time

#BCM. 23, wPi. 4, Physical. 16
FAN_IA = 23

GPIO.setmode(GPIO.BCM)
GPIO.setup(FAN_IA, GPIO.OUT)

pwm = GPIO.PWM(FAN_IA, 60)
pwm.start(0)

# weak
def weak() :
    pwm.ChangeDutyCycle(66)
    print("weak")
    time.sleep(5)

# medium
def Medium() :
    pwm.ChangeDutyCycle(33)
    print("Medium")
    time.sleep(5)

# strong
def Strong() :
    pwm.ChangeDutyCycle(0)
    print("Strong")
    time.sleep(5)


if __name__ == "__main__":
    try :
        while True:
            weak()
            Medium()
            Strong()
    except:
        GPIO.cleanup()
        print("end")
        